src/plonk/circuit.rs

```diff
@@ -6,21 +6,27 @@ use super::Error;
 use crate::arithmetic::Field;
 
 use crate::poly::Rotation;
-/// This represents a wire which has a fixed (permanent) value
+
+/// Enum over wires with variable values
 #[derive(Copy, Clone, Debug, Eq, PartialEq, Hash)]
-pub struct FixedWire(pub usize);
+pub enum Wire {
```

Comment:
I don't think we should do this. I think we should have a separate type `AuxWire` and in any APIs that need to be able to take in `AuxWire`/`AdviceWire`/`FixedWire` we can have a separate enum type `Wire` which implements `From<AuxWire>/From<AdviceWire>/From<FixedWire>`. In other cases we should be specific about the type.